tfx_bsl/beam/run_inference.py

```diff
@@ -697,81 +785,111 @@ def setup(self):
+        predict_log = prediction_log_pb2.PredictLog()
+        predict_log.CopyFrom(predict_log_tmpl)
+        if len(input_tensor_alias) == 1:
+          alias = input_tensor_alias[0]
+          predict_log.request.inputs[alias].string_val.append(process_elements[i])
+        else:
+          for alias, tensor_name in zip(input_tensor_alias, input_tensor_names):
+            predict_log.request.inputs[alias].float_val.append(elements[tensor_name][i])
 
-    result = []
-    for i in range(batch_size):
-      predict_log = prediction_log_pb2.PredictLog()
-      predict_log.CopyFrom(predict_log_tmpl)
-      predict_log.request.inputs[input_tensor_alias].string_val.append(
-          elements[i].SerializeToString())
-      for output_alias, output in outputs.items():
-        # Mimic tensor::Split
-        tensor_proto = tf.make_tensor_proto(
-            values=output[i],
-            dtype=tf.as_dtype(output[i].dtype).as_datatype_enum,
-            shape=np.expand_dims(output[i], axis=0).shape)
-        predict_log.response.outputs[output_alias].CopyFrom(tensor_proto)
-      result.append(predict_log)
+    for output_alias, output in outputs.items():
+      # Mimic tensor::Split
+      tensor_proto = tf.make_tensor_proto(
+          values=output[i],
+          dtype=tf.as_dtype(output[i].dtype).as_datatype_enum,
+          shape=np.expand_dims(output[i], axis=0).shape)
+      predict_log.response.outputs[output_alias].CopyFrom(tensor_proto)
+    result.append(predict_log)
     return result
 
 
-@beam.typehints.with_input_types(List[Union[tf.train.Example,
-                                            tf.train.SequenceExample]])
-@beam.typehints.with_output_types(Tuple[tf.train.Example,
-                                        inference_pb2.MultiInferenceResponse])
+@beam.typehints.with_input_types(pa.RecordBatch)
+@beam.typehints.with_output_types(Tuple[Union[str, bytes],
+                                  inference_pb2.MultiInferenceResponse])
 class _BatchMultiInferenceDoFn(_BaseBatchSavedModelDoFn):
   """A DoFn that runs inference on multi-head model."""
 
-  def _check_elements(
-      self, elements: List[Union[tf.train.Example,
-                                 tf.train.SequenceExample]]) -> None:
-    if not all(isinstance(element, tf.train.Example) for element in elements):
-      raise ValueError('Multi inference only supports tf.train.Example')
+  def _check_elements(self) -> None:
+    if self._data_type != DataType.EXAMPLE:
+      raise ValueError('Multi-inference only supports tf.train.Example')
 
   def _post_process(
-      self, elements: Sequence[tf.train.Example], outputs: Mapping[Text,
-                                                                   np.ndarray]
-  ) -> Iterable[Tuple[tf.train.Example, inference_pb2.MultiInferenceResponse]]:
+      self, elements: Mapping[Any, Any],
+      outputs: Mapping[Text, np.ndarray]
+  ) -> Iterable[Tuple[Union[str, bytes], inference_pb2.MultiInferenceResponse]]:
```

Comment:
Can this just be bytes instead of Union[str, bytes] ?
str is the same as 'bytes' in py2.

Reply:
Just wanted to make sure it's compatible with py2